.github/workflows/ci.yml

```diff
@@ -57,6 +57,7 @@ jobs:
           install: |
             apt-get update
             apt-get install -y --no-install-recommends python3 python3-pip
+            apt build-dep numpy
```

Comment:
Original error on `s390x` build at https://github.com/weiji14/cog3pio/actions/runs/8071580043/job/22051501437#step:7:410 before this line was added:

```python-traceback
    error: subprocess-exited-with-error
    
    × Preparing metadata (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> [20 lines of output]
        + /usr/bin/python3 /tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b/vendored-meson/meson/meson.py setup /tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b /tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b/.mesonpy-q6jzljl7 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b/.mesonpy-q6jzljl7/meson-python-native-file.ini
        The Meson build system
        Version: 1.2.99
        Source dir: /tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b
        Build dir: /tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b/.mesonpy-q6jzljl7
        Build type: native build
        Project name: NumPy
        Project version: 1.26.4
        
        ../meson.build:1:0: ERROR: Unknown compiler(s): [['cc'], ['gcc'], ['clang'], ['nvc'], ['pgcc'], ['icc'], ['icx']]
        The following exception(s) were encountered:
        Running `cc --version` gave "[Errno 2] No such file or directory: 'cc'"
        Running `gcc --version` gave "[Errno 2] No such file or directory: 'gcc'"
        Running `clang --version` gave "[Errno 2] No such file or directory: 'clang'"
        Running `nvc --version` gave "[Errno 2] No such file or directory: 'nvc'"
        Running `pgcc --version` gave "[Errno 2] No such file or directory: 'pgcc'"
        Running `icc --version` gave "[Errno 2] No such file or directory: 'icc'"
        Running `icx --version` gave "[Errno 2] No such file or directory: 'icx'"
        
        A full log can be found at /tmp/pip-install-6l_88pqi/numpy_281cd73ebd0a445c8bd7bbbb7479992b/.mesonpy-q6jzljl7/meson-logs/meson-log.txt
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: metadata-generation-failed
  
  × Encountered error while generating package metadata.
  ╰─> See above for output.
  
  note: This is an issue with the package mentioned above, not pip.
  hint: See above for details.
  25h::error::The process '/home/runner/work/_actions/uraimo/run-on-arch-action/517085f0367c8256bcfa753e3e13e1550af09954/src/run-on-arch.sh' failed with exit code 1
```

So at b54ab2644dc83ec61b15dc33a4d274138e153e59, I added this `apt build-dep numpy` line following https://numpy.org/devdocs/building/#system-level-dependencies, but got `E: You must put some 'deb-src' URIs in your sources.list` at https://github.com/weiji14/cog3pio/actions/runs/8071955898/job/22052723659?pr=3#step:7:330.

Maybe best to just install the compilers individually instead of messing with `/etc/apt/sources.list`

Reply:
Ok, added the NumPy build deps manually at 41259a1b337b20c15939833436051bb0a2484538, and also needed to `apt install ninja-build` to get NumPy to compile on armv7/s390x/ppc64le (9dfa03a6824dc23fc41ce345b04ff2f32d49a6fe). The CI on those three platforms take ~24min to run though, compared to <5min for linux/aarch64 :upside_down_face: